### PR TITLE
Hyperplonk PoC

### PR DESCRIPTION
**Pull request by alxkzmn**
This PoC doesn't yet have the user inclusion proof.

### CODE REVIEW

**Review by sifnoc — CHANGES_REQUESTED** [review]
Thanks for creating the PoC based on HyperPlonk!
I've left a few comments regarding some minor things I found.

File: prover/src/circuits/summa_circuit.rs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    {
+        type Config = SummaConfig<N_CURRENCIES, N_USERS>;
+        type FloorPlanner = SimpleFloorPlanner;
+
+        fn without_witnesses(&self) -> Self {
+            unimplemented!()
+        }
+
+        fn configure(meta: &mut ConstraintSystem<Fp>) -> Self::Config {
+            //TODO what's the meaning of this?
```
[the remark this answers is not in the document]
Comment (alxkzmn) [reply prover/src/circuits/summa_circuit.rs:128]: I see, thanks! I wonder if it affects the performance in any way.

File: prover/benches/proof_of_liabilities.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    let mut c = Criterion::default().sample_size(10);
+
+    let grand_sum_proof_bench_name = format!("<{}> grand sum proof", name);
+
+    type Pb = HyperPlonk<MultilinearKzg<Bn256>>;
+    let entries = generate_dummy_entries::<N_USERS, N_CURRENCIES>().unwrap();
+    let halo2_circuit = SummaHyperplonk::<N_USERS, N_CURRENCIES>::init(entries.to_vec());
+
+    let circuit = Halo2Circuit::<Fp, SummaHyperplonk<N_USERS, N_CURRENCIES>>::new::<Pb>(
+        17,
```
[the remark this answers is not in the document]
Comment (alxkzmn) [reply prover/benches/proof_of_liabilities.rs:31]: Sure

**Review by sifnoc — APPROVED** [review]
Looks good to me!

**Pull request merged**